consensus/ibft/consensus_backend.go

```diff
@@ -137,20 +128,26 @@ func (i *backendIBFT) buildBlock(snap *Snapshot, parent *types.Header) (*types.B
 
 	header.GasLimit = gasLimit
 
-	if hookErr := i.runHook(CandidateVoteHook, header.Number, &candidateVoteHookParams{
-		header: header,
-		snap:   snap,
-	}); hookErr != nil {
-		i.logger.Error(fmt.Sprintf("Unable to run hook %s, %v", CandidateVoteHook, hookErr))
+	if err := i.currentHooks.ModifyHeader(header, i.currentSigner.Address()); err != nil {
+		return nil, err
 	}
 
 	// set the timestamp
 	header.Timestamp = uint64(time.Now().Unix())
 
-	// we need to include in the extra field the current set of validators
-	putIbftExtraValidators(header, snap.Set)
+	var parentCommittedSeal signer.Sealer
 
-	transition, err := i.executor.BeginTxn(parent.StateRoot, header, i.validatorKeyAddr)
+	if header.Number > 1 {
+		if parentCommittedSeal, err = i.extractCommittedSeals(parent); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := i.currentSigner.InitIBFTExtra(header, parentCommittedSeal, i.currentValidators); err != nil {
+		return nil, err
+	}
```

Comment:
What do you think about putting this behind a helper?
The `buildBlock` method is bloated enough as it is

Reply:
Added a helper method, 06cd4f01dfe4a31a92d92e5219670a4e4d71a93c